src/layout/page-layout.scss

```diff
@@ -0,0 +1,384 @@
+    }
+  }
+
+  // outer spacing
+
+  &.PageLayout--outerSpacing-normal {
+    --Layout-outer-spacing-x: #{$spacer-3}; // 16px
+    --Layout-outer-spacing-y: #{$spacer-3}; // 16px
+
+    @include breakpoint(lg) {
+      --Layout-outer-spacing-x: #{$spacer-4}; // 24px
+      --Layout-outer-spacing-y: #{$spacer-4}; // 24px
+    }
+  }
+
+  &.PageLayout--outerSpacing-condensed {
+    --Layout-outer-spacing-x: #{$spacer-3}; // 16px
+    --Layout-outer-spacing-y: #{$spacer-3}; // 16px
+  }
+
+  // inner spacing
+
+  &.PageLayout--innerSpacing-normal {
+    --Layout-inner-spacing-min: #{$spacer-3}; // 16px
+    --Layout-inner-spacing-max: #{$spacer-3}; // 16px
+
+    @include breakpoint(lg) {
+      --Layout-inner-spacing-max: #{$spacer-4}; // 24px
+    }
+  }
+
+  &.PageLayout--innerSpacing-condensed {
+    --Layout-inner-spacing-min: #{$spacer-3}; // 16px
+    --Layout-inner-spacing-max: #{$spacer-3}; // 16px
+  }
+
+  // column gap
+
+  &.PageLayout--columnGap-normal {
+    --Layout-column-gap: #{$spacer-3}; // 16px
+
+    @include breakpoint(lg) {
+      --Layout-column-gap: #{$spacer-4}; // 24px
+    }
+  }
+
+  &.PageLayout--columnGap-condensed {
+    --Layout-column-gap: #{$spacer-3}; // 16px
+  }
+
+  &.PageLayout--columnGap-none {
+    // the `px` unit is mandatory for `calc()` execution. See https://stackoverflow.com/a/32518348
+    // stylelint-disable-next-line length-zero-no-unit
+    --Layout-column-gap: 0px;
+  }
+
+  // row gap
+
+  &.PageLayout--rowGap-normal {
+    --Layout-row-gap: #{$spacer-3}; // 16px
+
+    @include breakpoint(lg) {
+      --Layout-row-gap: #{$spacer-4}; // 24px
+    }
+  }
+
+  &.PageLayout--rowGap-none {
+    // the `px` unit is mandatory for `calc()` execution. See https://stackoverflow.com/a/32518348
+    // stylelint-disable length-zero-no-unit
+    --Layout-row-gap: 0px;
+  }
+
+  &.PageLayout--rowGap-condensed {
+    --Layout-row-gap: #{$spacer-3}; // 16px
+  }
+
+  // regions
+
+  .PageLayout-header,
+  .PageLayout-content,
+  .PageLayout-pane,
+  .PageLayout-footer {
+    // stylelint-disable-next-line primer/spacing
+    padding: var(--Layout-inner-spacing-min);
+  }
+
+  .PageLayout-content {
+    // stylelint-disable primer/spacing
+    padding-right: var(--Layout-inner-spacing-max);
+    padding-left: var(--Layout-inner-spacing-max);
+    // stylelint-enable primer/spacing
+    grid-area: content;
+  }
+
+  .PageLayout-pane {
+    grid-area: pane;
+  }
+
+  // multi-column desktop-friendly layout
+
+  @include breakpoint($Layout-responsive-variant-max-breakpoint) {
+
+    // Set a `content` region width, to work with loading states when
+    // `pane` is not yet loaded. See https://github.com/primer/css/pull/1818
+    $Layout-content-full-width: minmax(0, calc(100% - var(--Layout-pane-width) - var(--Layout-column-gap)));
+
+    &.PageLayout--panePos-start {
+      --Layout-template-columns: var(--Layout-pane-width) #{$Layout-content-full-width};
+      --Layout-template-areas: 'pane content';
```

Comment:
> I would suggest we handle this in the PVC/PRC component to ensure the content is rendered in the correct order (maybe you were already expecting this behavior, not sure!)

Right! This is how I'm expecting the PVC/PRC components to handle it, too. The pane position modifier class should match with the markup rendering, having the `pane` region rendered first if `position` is set to `start`, and vice versa. [This is how Storybook](https://github.com/primer/css/pull/1737/files#diff-071e314867fe1d80ffa58a7abb8969946d7972861b52acd529e3fc78dceed7dfR376-R385) is handling and [how the PVC component is doing it](https://github.com/primer/view_components/pull/937/files#diff-a48d7eea76b913487f1bd98456e2b17f04f65bb502951d7d9c7c1c9ef975c9e6R5-R11).

However! If `PageLayout` has its responsive variant set to `stackRegions`, I've considered a prop called `responsivePosition` in the `pane` slot that allows for customizing the `pane` position on smaller viewports. Its default value is `inherit`, which should match with the larger screen/DOM ordering, and wouldn't be an accessibility issue.

But in case `responsivePosition` is set to something else, that would break it indeed 😬 . I was using [this mockup](https://www.figma.com/file/fWoXP4StfljyBfYiOhlPaJ/2020-10-Responsive-explorations?node-id=106%3A20720) (internal to hubbers) as a reference of a pane that on larger screens has its position set to `end`, but with `stackRegions` gets a responsive position set to `start`.

I'll try to find some more examples of how `stackRegions` would work with the current pages to see if this prop is really needed. :v:

---

I do wonder how fragile it'd be to move the DOM nodes on `resize` 😬 😬 😬, with something like [`insertBefore`](https://developer.mozilla.org/en-US/docs/Web/API/Node/insertBefore) (which can be used in elements that currently exist in the document).

Reply:
@ichelsea can you explain the problem in your example?

Using your example, I don't understand why would I want side content to be read first on mobile but not on desktop. It's the same content, just a different screen size.

Reply:
> I do wonder how fragile it'd be to move the DOM nodes on `resize` 😬 😬 😬, with something like `insertBefore` (which can be used in elements that currently exist in the document).

I wanted to explore how feasible it is to reorder the DOM elements on demand, and how performant is is to use `insertBefore` to reflect the position of the columns as they take place depending on the responsive behavior.

If this makes any sense in terms of accessibility, maybe it's something we should support as a [Primer behavior](https://github.com/primer/behaviors).

[Codepen playground](https://codepen.io/team/GitHub/pen/abLWOWQ)

c/c @ichelsea @alliethu @smockle

Reply:
@mperrotti 

> Using your example, I don't understand why would I want side content to be read first on mobile but not on desktop. It's the same content, just a different screen size.

You're right, you wouldn't! The DOM order should be read however it's visually presented; I might not have been super clear on that, apologies! The example was how it shouldn't be done 😅